src/Agent.php

```diff
@@ -64,6 +64,64 @@ public function asVersion($version) {
         return $result;
     }
 
+    public function isIdentified() {
+        return (isset($this->mbox) || isset($this->mbox_sha1sum) || isset($this->openid) || isset($this->account));
+    }
+
+    //
+    // having multiple IFIs set shouldn't cause a problem here
+    // so long as all of their values match their counterparts
+    //
+    // we could allow for multiple IFIs in the `this` object and
+    // ignore them missing in the signature but discussion ruled
+    // against that need since the serialization via asVersion
+    // shouldn't result in that ever for a valid statement
+    //
+    public function compareWithSignature($fromSig) {
+        //
+        // mbox and mbox_sha1sum are a special case where they can be
+        // equal but have to be transformed for comparison, check them
+        // first and short circuit
+        //
+        if (isset($this->mbox) && isset($fromSig->mbox_sha1sum)) {
+            if ($this->getMbox_sha1sum() === $fromSig->mbox_sha1sum) {
+                return array('success' => true, 'reason' => null);
+            }
+
+            return array('success' => false, 'reason' => 'Comparison of this.mbox to signature.mbox_sha1sum failed: no match');
+        }
+        else if (isset($fromSig->mbox) && isset($this->mbox_sha1sum)) {
+            if ($fromSig->getMbox_sha1sum() === $this->mbox_sha1sum) {
+                return array('success' => true, 'reason' => null);
+            }
+
+            return array('success' => false, 'reason' => 'Comparison of this.mbox_sha1sum to signature.mbox failed: no match');
+        }
+
+        foreach (array('mbox', 'mbox_sha1sum', 'openid') as $property) {
+            if (isset($this->$property) || isset($fromSig->$property)) {
+                if ($this->$property !== $fromSig->$property) {
+                    return array('success' => false, 'reason' => "Comparison of $property failed: value is not the same");
+                }
+            }
+        }
+        if (isset($this->account) || isset($fromSig->account)) {
+            if (! isset($fromSig->account)) {
+                return array('success' => false, 'reason' => "Comparison of $property failed: value not in signature");
+            }
+            if (! isset($this->account)) {
+                return array('success' => false, 'reason' => "Comparison of $property failed: value not in this");
```

Comment:
Possible improvement: it might be nice to say what "this" is each time (in this case is an Agent). There's only 8 occurrences of "not in this".